src/state/slices/assetsSlice/selectors.ts

```diff
@@ -21,17 +21,23 @@ import createCachedSelector from 're-reselect'
 import { isSome } from 'lib/utils'
 import type { ReduxState } from 'state/reducer'
 import { createDeepEqualOutputSelector } from 'state/selector-utils'
+import { selectAssetIdParamFromFilter } from 'state/selectors'
 import { selectCryptoMarketDataIdsSortedByMarketCap } from 'state/slices/marketDataSlice/selectors'
 
 import { assetIdToFeeAssetId } from '../portfolioSlice/utils'
 
 export const selectAssetById = createCachedSelector(
   (state: ReduxState) => state.assets.byId,
   (_state: ReduxState, assetId: AssetId) => assetId,
-  // TODO(0xdef1cafe): make this return type AssetId | undefined and fix the 600+ type errors
   (byId, assetId) => byId[assetId] || undefined,
 )((_state: ReduxState, assetId: AssetId | undefined): AssetId => assetId ?? 'undefined')
 
+export const selectAssetByFilter = createCachedSelector(
```

Comment:
The same as above but can be used with `byFilter` selectors, something something arity